lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: MistralAIModel.values
+            .map(
+              (model) => RadioListTile<MistralAIModel>(
+                title: Text(
+                  model.name,
+                  maxLines: 1,
+                ),
+                value: model,
+                controlAffinity: ListTileControlAffinity.trailing,
+                groupValue: selectedModel,
+                onChanged: (value) {
+                  if (value == null) return;
+                  setState(() => selectedModel = value);
+                },
+              ),
+            )
+            .toList(),
+      ),
+    );
+  }
+}
+
+class MaxTokensDialog extends StatefulWidget {
+  const MaxTokensDialog({
+    super.key,
+  });
+
+  @override
+  State<MaxTokensDialog> createState() => _MaxTokensDialogState();
+}
+
+class _MaxTokensDialogState extends State<MaxTokensDialog> {
+  final TextEditingController _maxTokensController = TextEditingController();
+
+  @override
+  void initState() {
+    _maxTokensController.text =
+        context.read<SummarySettingsModel>().settings.maxTokens?.toString() ??
+            '';
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _maxTokensController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Max Tokens',
+      description: 'Write a number if you want to set a max of tokens',
+      onSettingChanged: () {
+        final maxTokens = _maxTokensController.text.isEmpty
+            ? null
+            : int.tryParse(_maxTokensController.text);
+        context.read<SummarySettingsModel>().setMaxTokens(maxTokens);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _maxTokensController,
+            onChanged: (value) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_maxTokensController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 500',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unlimited',
+              ),
+              Switch(
+                value: _maxTokensController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class RandomSeedDialog extends StatefulWidget {
```

Comment:
This dialog is almost the same as `MaxTokensDialog`. Probably you could create a single dialog for both cases.